app/assets/javascripts/discourse/tests/integration/components/async-content-test.gjs

```diff
@@ -0,0 +1,350 @@
+      const promise = async () => "data";
+
+      await render(<template>
+        <AsyncContent @asyncData={{promise}}>
+          <:content as |data|>
+            <div class="content">{{data}}</div>
+          </:content>
+        </AsyncContent>
+      </template>);
+
+      assert.dom(".content").hasText("data");
+    });
+
+    test("it accepts an instance of TrackedAsyncData", async function (assert) {
+      const promise = new TrackedAsyncData(Promise.resolve("data"));
+
+      await render(<template>
+        <AsyncContent @asyncData={{promise}}>
+          <:content as |data|>
+            <div class="content">{{data}}</div>
+          </:content>
+        </AsyncContent>
+      </template>);
+
+      assert.dom(".content").hasText("data");
+    });
+  });
+
+  module("@context", function () {
+    test("it passes the context to the async function", async function (assert) {
+      const promise = (context) => {
+        assert.strictEqual(context, "correct", "context is passed correctly");
+        return Promise.resolve("data");
+      };
+
+      await render(<template>
+        <AsyncContent @asyncData={{promise}} @context="correct">
+          <:content as |data|>
+            <div class="content">{{data}}</div>
+          </:content>
+        </AsyncContent>
+      </template>);
+
+      assert.dom(".content").hasText("data");
+    });
+
+    test("it updates the async data when the context changes", async function (assert) {
+      await render(
+        class extends Component {
+          @tracked context = "first";
+
+          @action
+          changeContext() {
+            this.context = "second";
+          }
+
+          async load(context) {
+            return context;
+          }
+
+          <template>
+            <button {{on "click" this.changeContext}}>Change Context</button>
+            <AsyncContent @asyncData={{this.load}} @context={{this.context}}>
+              <:content as |data|>
+                <div class="content">{{data}}</div>
+              </:content>
+            </AsyncContent>
+          </template>
+        }
+      );
+
+      assert.dom(".content").hasText("first");
+
+      await click("button");
+
+      assert.dom(".content").hasText("second");
+    });
+  });
+
+  module("<:loading>", function () {
+    test("it displays the spinner when the block is not provided", async function (assert) {
+      const promise = new Promise((resolve) => {
+        setTimeout(() => {
+          assert.dom(".spinner").exists();
+          resolve();
+        }, 250);
+      });
+
+      await render(<template>
+        <AsyncContent @asyncData={{promise}}>
+          <:content>
+            <div class="content"></div>
+          </:content>
+        </AsyncContent>
+      </template>);
+
+      assert.dom(".content").exists();
+    });
+
+    test("it displays the block when provided", async function (assert) {
+      const promise = new Promise((resolve) => {
+        setTimeout(() => {
```

Comment:
As above, I think we should avoid the setTimeout

Reply:
Replaced with the pattern recommended in the `ember-async-data` README